### PR TITLE
Support groupBy:year and groupBy:quarter in the App

### PR DESCRIPTION
**Pull request by cristipaval**
### Fixed Issues

$ https://github.com/Expensify/App/issues/80397
$ https://github.com/Expensify/App/issues/80398
$ https://github.com/Expensify/App/issues/80849
PROPOSAL:

### Prerequisites:
- Ensure you have expenses with dates spanning multiple years (e.g., expenses from 2024, 2025, 2026)
- Ensure you have expenses spanning multiple quarters (e.g., Q1, Q2, Q3, Q4 of the same year, and across different years)
- Ensure you have expenses from different months within quarters (e.g., January, February, March for Q1)

---

### Test 1: Access Group by Year/Quarter via Advanced Filters

**For Year:**
1. Open the app and navigate to the Reports page
2. Go to Expense Section
3. Tap the filter in the search header
4. Tap on "Group by" option
5. Verify "Year" appears in the list of group by options
6. Select "Year"
7. Tap "Save"
8. Verify expenses are now grouped by year with headers showing:
   - Year (e.g., "2026", "2025", "2024")
   - Number of expenses in that year
   - Total amount for that year

**For Quarter:**
1. Open the app and navigate to the Reports page
2. Go to Expense Section
3. Tap the filter in the search header
4. Tap on "Group by" option
5. Verify "Quarter" appears in the list of group by options
6. Select "Quarter"
7. Tap "Save"
8. Verify expenses are now grouped by quarter with headers showing:
   - Quarter label (e.g., "Q1 2026", "Q2 2026", "Q4 2025")
   - Number of expenses in that quarter
   - Total amount for that quarter

---

### Test 2: Verify Year/Quarter Grouping Display (Large Screen)

**For Year:**
1. Open the app on desktop/web or tablet in landscape mode
2. Go to Expense Section
3. Navigate to Search and apply `groupBy:year` filter
4. Verify the table header shows: Year icon | Year | Expenses | Total
5. Verify each year row displays the calendar icon, year (e.g., "2026"), expense count, and total amount
6. Verify columns are properly aligned between header and rows

**For Quarter:**
1. Open the app on desktop/web or tablet in landscape mode
2. Go to Expense Section
3. Navigate to Search and apply `groupBy:quarter` filter
4. Verify the table header shows: Quarter icon | Quarter | Expenses | Total
5. Verify each quarter row displays the calendar icon, quarter label (e.g., "Q1 2026"), expense count, and total amount
6. Verify columns are properly aligned between header and rows

---

### Test 3: Verify Year/Quarter Grouping Display (Small Screen)

**For Year:**
1. Open the app on mobile device
2. Go to Expense Section
3. Navigate to Search and apply `groupBy:year` filter
4. Verify each year shows the calendar icon, year label, and total amount
5. Tap the expand arrow on a year row
6. Verify the individual transactions within that year are displayed

**For Quarter:**
1. Open the app on mobile device
2. Go to Expense Section
3. Navigate to Search and apply `groupBy:quarter` filter
4. Verify each quarter shows the calendar icon, quarter label (e.g., "Q1 2026"), and total amount
5. Tap the expand arrow on a quarter row
6. Verify the individual transactions within that quarter are displayed

---

### Test 4: Verify Year/Quarter Grouping Across Multiple Periods

**For Year:**
1. Create expenses with dates spanning at least 3 different years (e.g., 2024, 2025, 2026)
2. Go to Expense Section
3. Navigate to Search with `groupBy:year`
4. Verify expenses are grouped into separate year sections
5. Verify each year section shows the correct year label
6. Verify transactions appear only in their respective year groups

**For Quarter:**
1. Create expenses with dates spanning at least 3 different quarters (e.g., Q1 2026, Q2 2026, Q4 2025)
2. Go to Expense Section
3. Navigate to Search with `groupBy:quarter`
4. Verify expenses are grouped into separate quarter sections
5. Verify each quarter section shows the correct quarter label (e.g., "Q1 2026", "Q2 2026")
6. Verify transactions appear only in their respective quarter groups

---

### Test 5: Verify Year/Quarter Expand/Collapse

**For Year:**
1. Navigate to Search with `groupBy:year` applied
2. Tap/click on a year row (e.g., "2026")
3. Verify the year row expands to show individual transactions within that year
4. Tap/click on the year row again
5. Verify the year row collapses to hide the transactions

**For Quarter:**
1. Navigate to Search with `groupBy:quarter` applied
2. Tap/click on a quarter row (e.g., "Q1 2026")
3. Verify the quarter row expands to show individual transactions within that quarter
4. Tap/click on the quarter row again
5. Verify the quarter row collapses to hide the transactions

---

### Test 6: Verify Sorting

**For Year:**
1. Navigate to Search with `groupBy:year`
2. Verify years are sorted chronologically (newest first by default, e.g., "2026", "2025", "2024")
3. Click on the "Year" column header to sort
4. Verify years sort chronologically (oldest first, then newest first on second click)
5. Click on the "Total" column header
6. Verify years sort by total amount (descending, then ascending on second click)

**For Quarter:**
1. Navigate to Search with `groupBy:quarter`
2. Verify quarters are sorted chronologically (newest first by default, e.g., "Q1 2026", "Q4 2025", "Q3 2025")
3. Click on the "Quarter" column header to sort
4. Verify quarters sort chronologically (oldest first, then newest first on second click)
5. Click on the "Total" column header
6. Verify quarters sort by total amount (descending, then ascending on second click)

---

### Test 7: Verify Multi-select Checkbox

**For Year:**
1. Navigate to Search with `groupBy:year`
2. Enable selection mode (long press on mobile, or use bulk select)
3. Tap the checkbox on a year header row
4. Verify all transactions within that year become selected

**For Quarter:**
1. Navigate to Search with `groupBy:quarter`
2. Enable selection mode (long press on mobile, or use bulk select)
3. Tap the checkbox on a quarter header row
4. Verify all transactions within that quarter become selected

---

### Test 8: Verify Quarter Grouping Across Year Boundaries

**For Quarter:**
1. Create expenses with dates spanning Q4 2025 (October-December 2025) and Q1 2026 (January-March 2026)
2. Go to Expense Section
3. Navigate to Search with `groupBy:quarter`
4. Verify expenses are grouped into separate quarter sections (e.g., "Q1 2026", "Q4 2025")
5. Verify transactions appear only in their respective quarter groups
6. Verify quarters are sorted correctly across year boundaries (newest first, so Q1 2026 appears before Q4 2025)

---

### Test 9: Verify Quarter Boundary Transactions (Critical Edge Case)

**For Quarter:**
1. Create expenses on the first day of a quarter (e.g., January 1, 2026 for Q1)
2. Create expenses on the last day of a quarter (e.g., March 31, 2026 for Q1)
3. Create expenses on the day before quarter start (e.g., December 31, 2025)
4. Create expenses on the day after quarter end (e.g., April 1, 2026)
5. Go to Expense Section
6. Navigate to Search with `groupBy:quarter`
7. Verify expenses from January 1 - March 31 appear in Q1 2026 group
8. Verify expenses from December 31, 2025 appear in Q4 2025 group
9. Verify expenses from April 1, 2026 appear in Q2 2026 group
10. Verify transactions are NOT incorrectly grouped across quarter boundaries

---

### Test 10: Verify Clicking Year/Quarter/Week Group Filters Transactions

**For Year:**
1. Navigate to Search with `groupBy:year`
2. Click/tap on a year group header (e.g., "2026")
3. Verify the search query updates to filter by that year's date range (e.g., "2026-01-01" to "2026-12-31")
4. Verify `groupBy` is removed from the query
5. Verify only transactions from that year are displayed as individual items (not grouped)
6. Verify the date filter shows the year's start and end dates

**For Quarter:**
1. Navigate to Search with `groupBy:quarter`
2. Click/tap on a quarter group header (e.g., "Q1 2026")
3. Verify the search query updates to filter by that quarter's date range (e.g., "2026-01-01" to "2026-03-31" for Q1)
4. Verify `groupBy` is removed from the query
5. Verify only transactions from that quarter are displayed as individual items (not grouped)
6. Verify the date filter shows the quarter's start and end dates

**For Week:**
1. Navigate to Reports > Expense
2. Enter in the Search box field: 'groupBy:week'
3. Click on a week group header (e.g., "Jan 25 - Jan 31, 2026")
4. Verify the search query updates to filter by that week's date range
5. Verify groupBy is removed from the query
6. Verify only transactions from that week are displayed as individual items (not grouped)

---

### Test 11: Verify Year-to-Date Filter with Year/Quarter Grouping

**For Year:**
1. Navigate to Search and apply `date:year-to-date` filter
2. Apply `groupBy:year` filter
3. Verify only the current year appears in the results
4. Verify transactions from previous years are filtered out

**For Quarter:**
1. Navigate to Search and apply `date:year-to-date` filter
2. Apply `groupBy:quarter` filter
3. Verify only quarters from the current year appear in the results (e.g., if current date is March 15, 2026, only Q1 2026 should appear)
4. Verify transactions from previous years' quarters are filtered out
5. Click on a quarter group (e.g., "Q1 2026")
6. Verify the date filter correctly intersects with year-to-date (only shows transactions from Q1 2026, not from previous year)

---

### Test 12: Verify Quarter Formatting and Display

**For Quarter:**
1. Navigate to Search with `groupBy:quarter`
2. Verify quarter labels are formatted correctly:
   - Q1 2026 (January-March)
   - Q2 2026 (April-June)
   - Q3 2026 (July-September)
   - Q4 2026 (October-December)
3. Verify quarter labels show both the quarter number and year
4. Verify quarters from different years are clearly distinguishable (e.g., "Q1 2026" vs "Q1 2025")

---

- [x] Verify that no errors appear in the JS console

### QA Steps

Same as tests.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified there are no new alerts related to the `canBeMissing` param for `useOnyx`
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4510fc76bbf5df699a2575bfb49a276af90f3ed7/src/components/LocaleContextProvider.tsx#L80)
      - [x] If any non-english text was added/modified, I used [JaimeGPT](https://chatgpt.com/g/g-2dgOQl5VM-english-to-spanish-translator-aka-jaimegpt) to get English > Spanish translation. I then posted it in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4510fc76bbf5df699a2575bfb49a276af90f3ed7/src/components/LocaleContextProvider.tsx#L116-L123)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.ts or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If new assets were added or existing ones were modified, I verified that:
    - [x] The assets are optimized and compressed (for SVG files, run `npm run compress-svg`)
    - [x] The assets load correctly across all supported platforms.
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I added [unit tests](https://github.com/Expensify/App/blob/main/tests/README.md) for any new feature or bug fix in this PR to help automatically prevent regressions in this user flow.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I verified that similar component doesn't exist in the codebase
- [x] I verified that all props are defined accurately and each prop has a `/** comment above it */`
- [x] I verified that each file is named correctly
- [x] I verified that each component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
- [x] I verified that the only data being stored in component state is data necessary for rendering and nothing else
- [x] In component if we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
- [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
- [x] I verified that component internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
- [x] I verified that all JSX used for rendering exists in the render method
- [x] I verified that each component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/bb6e7a65-8ca8-41d0-bf57-e7ac6983d66c

</details>

### CODE REVIEW

**Review by trjExpensify — DISMISSED** [review]
Core PR for a WN project, no qualms from me! 👍

File: src/components/Search/index.tsx [hunk 1]
```diff
@@ -930,7 +932,62 @@ function Search({
...
+                    return;
+                }
+                const newFlatFilters = queryJSON.flatFilters.filter((filter) => filter.key !== CONST.SEARCH.SYNTAX_FILTER_KEYS.DATE);
+                // Calculate quarter date range: Q1 (Jan-Mar), Q2 (Apr-Jun), Q3 (Jul-Sep), Q4 (Oct-Dec)
+                const quarterStartMonth = (quarterGroupItem.quarter - 1) * 3 + 1; // 1, 4, 7, 10
+                const quarterEndMonth = quarterGroupItem.quarter * 3; // 3, 6, 9, 12
+                const quarterStart = `${quarterGroupItem.year}-${String(quarterStartMonth).padStart(2, '0')}-01`;
+                // Get last day of quarter end month (Date constructor uses 0-indexed months, so quarterEndMonth gives last day of previous month)
+                const quarterEndDate = new Date(quarterGroupItem.year, quarterEndMonth, 0);
+                const quarterEnd = `${quarterGroupItem.year}-${String(quarterEndMonth).padStart(2, '0')}-${String(quarterEndDate.getDate()).padStart(2, '0')}`;
```
Comment (shubham1206agra) [inline src/components/Search/index.tsx]: Use `getQuarterDateRange` to obtain the range to maintain consistent logic.

File: src/components/SelectionListWithSections/Search/TransactionGroupListItem.tsx [hunk 2]
```diff
@@ -114,7 +114,8 @@ function TransactionGroupListItem<TItem extends ListItem>({
     const [isExpanded, setIsExpanded] = useState(false);
     const [isActionLoadingSet = new Set<string>()] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT_METADATA}`, {canBeMissing: true, selector: isActionLoadingSetSelector});
     const [bankAccountList] = useOnyx(ONYXKEYS.BANK_ACCOUNT_LIST, {canBeMissing: true});
-    const shouldOpenGroupOnPress = !isExpenseReportType && (groupBy === CONST.SEARCH.GROUP_BY.YEAR || groupBy === CONST.SEARCH.GROUP_BY.QUARTER);
+    const shouldOpenGroupOnPress =
+        !isExpenseReportType && (groupBy === CONST.SEARCH.GROUP_BY.YEAR || groupBy === CONST.SEARCH.GROUP_BY.QUARTER || groupBy === CONST.SEARCH.GROUP_BY.WEEK);
```
Comment (shubham1206agra) [inline src/components/SelectionListWithSections/Search/TransactionGroupListItem.tsx]: Is this not required for other groupBy?

Comment (lakchote) [reply src/components/SelectionListWithSections/Search/TransactionGroupListItem.tsx]: It's limited to week/year/quarter on purpose as those are the time-based groupings where row press should drill down to a date range, while other `groupBy` types should keep row press for expand/collapse.

Comment (shubham1206agra) [reply src/components/SelectionListWithSections/Search/TransactionGroupListItem.tsx]: @lakchote Then, can you please include month in this?

Comment (lakchote) [reply src/components/SelectionListWithSections/Search/TransactionGroupListItem.tsx]: This doesn't work out of the box for the groupBy month, I suspect because it's not a time interval that is passed 

image

Compared to other groups:

image

Since time is running by and we need to keep momentum high, I suggest to handle this in a follow-up since this PR concerns groupBy year and quarter.

Comment (lakchote) [reply src/components/SelectionListWithSections/Search/TransactionGroupListItem.tsx]: More on this [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1769697275626839?thread_ts=1769673521.116649&cid=C01GTK53T8Q)

**Review by shubham1206agra — APPROVED** [review]

**Review by neil-marcellini — CHANGES_REQUESTED** [review]
As seen in the test videos, expanding a quarter group by group navigates to a new view vs expanding in line as it should. Please fix that.

File: src/components/Search/index.tsx [hunk 3]
```diff
@@ -930,7 +932,74 @@ function Search({
...
                 }
-                handleSearch({queryJSON: newQueryJSONWithHash, searchKey, offset: 0, shouldCalculateTotals: false, isLoading: false});
+                setSearchContext(true);
+                Navigation.navigate(
+                    ROUTES.SEARCH_ROOT.getRoute({
+                        query: newQuery,
+                        rawQuery: undefined,
+                    }),
+                );
+                return;
```
Comment (neil-marcellini) [inline src/components/Search/index.tsx:940]: this is the problematic navigation. We should call handleSearch like the other group by conditions

**Review by neil-marcellini — CHANGES_REQUESTED** [review]

File: src/components/SelectionListWithSections/Search/TransactionGroupListItem.tsx [hunk 4]
```diff
@@ -110,6 +114,7 @@ function TransactionGroupListItem<TItem extends ListItem>({
     const [isExpanded, setIsExpanded] = useState(false);
     const [isActionLoadingSet = new Set<string>()] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT_METADATA}`, {canBeMissing: true, selector: isActionLoadingSetSelector});
     const [bankAccountList] = useOnyx(ONYXKEYS.BANK_ACCOUNT_LIST, {canBeMissing: true});
+    const shouldOpenGroupOnPress = !isExpenseReportType && (groupBy === CONST.SEARCH.GROUP_BY.YEAR || groupBy === CONST.SEARCH.GROUP_BY.QUARTER || groupBy === CONST.SEARCH.GROUP_BY.WEEK);
```
Comment (neil-marcellini) [inline src/components/SelectionListWithSections/Search/TransactionGroupListItem.tsx]: I don't understand this. I thought groupBy can't be set when viewing search for expense reports?

File: src/libs/SearchUIUtils.ts [hunk 5]
```diff
@@ -3093,9 +3223,13 @@ function getCustomColumns(value?: SearchDataTypes | SearchGroupBy): SearchCustom
         case CONST.SEARCH.GROUP_BY.TAG:
             return Object.values(CONST.SEARCH.GROUP_CUSTOM_COLUMNS.TAG);
         case CONST.SEARCH.GROUP_BY.MONTH:
-            return Object.values(CONST.SEARCH.GROUP_CUSTOM_COLUMNS.MONTH);
+            return Object.values(CONST.SEARCH.GROUP_CUSTOM_COLUMNS.MONTH) as SearchCustomColumnIds[];
```
Comment (neil-marcellini) [inline src/libs/SearchUIUtils.ts]: NAB: Why do we need this cast but not for the others?

File: tests/unit/Search/SearchUIUtilsTest.ts [hunk 6]
```diff
@@ -2777,6 +2887,170 @@ describe('SearchUIUtils', () => {
...
+                currentUserEmail: '',
+                translate: translateLocal,
+                formatPhoneNumber,
+                bankAccountList: {},
+                groupBy: CONST.SEARCH.GROUP_BY.QUARTER,
+            }) as [TransactionQuarterGroupListItemType[], number];
+
+            expect(result).toHaveLength(2);
+            expect(result.some((item) => item.formattedQuarter.startsWith('Q1 2026'))).toBe(true);
+            expect(result.some((item) => item.formattedQuarter.startsWith('Q3 2026'))).toBe(true);
```
Comment (neil-marcellini) [inline tests/unit/Search/SearchUIUtilsTest.ts]: NAB: this should probably assert an exact match

File: tests/unit/Search/SearchUIUtilsTest.ts [hunk 7]
```diff
@@ -1869,17 +1871,19 @@ const transactionMerchantGroupListItemsSorted: TransactionMerchantGroupListItemT
     },
 ];
 
-const searchResultsGroupByWeek: OnyxTypes.SearchResults = {
+const searchResultsGroupByMonth: OnyxTypes.SearchResults = {
```
Comment (neil-marcellini) [inline tests/unit/Search/SearchUIUtilsTest.ts:1874]: NAB: This follow existing patterns so it's ok, but for these data sets only used in one test case, it would be more readable to declare them inline.

**Review by neil-marcellini — COMMENTED** [review]
more thoughts...

File: src/components/SelectionListWithSections/Search/TransactionGroupListItem.tsx (hunk 4 above)
Comment (cristipaval) [reply src/components/SelectionListWithSections/Search/TransactionGroupListItem.tsx]: yeah, this is not needed. I guess it was added not to expand on click. But we want to expand.

File: src/libs/SearchUIUtils.ts (hunk 5 above)
Comment (cristipaval) [reply src/libs/SearchUIUtils.ts]: I guess AI added it and we didn't notice. It isn't needed.

**Review by neil-marcellini — DISMISSED** [review]
Thanks for the updates, it looks good now. Unfortunately some checks are failing.

**Review by neil-marcellini — APPROVED** [review]
